### ISSUE TITLE
Use phi to resolve values in loops

### ISSUE THREAD

**Issue by AjaniBilby** [qupa-lang/Qupa#38]
Source:
```
int main() {
	int i = 0;

	while (i < 10) {
		println(i);
		i = i + 1;
	}

	println(i);

	return 0;
}
```
Should produce (presuming latent allocation has been implemented #19):
```llvm
define dso_local i32 @main() #1 {
	br label %1

1:
	%2 = phi i32 [ 0, %0 ], [ %5, %4 ]
	%3 = icmp ult i32 %2, 10
	br i1 %3, label %4, label %6

4:
	call void @println.kyfn.g(i32 %2)
	%5 = add i32 %2, 1
	br label %1

6:
	%7 = phi i32 [ %2, %1 ], [ %5, %4 ]
	call void @println.kyfn.g(i32 %7)
	ret i32 0
}
```

**Comment by AjaniBilby** [comment]
Reduce the number of jumps and phi statements by checking once when entering the loop, and on each recursion.
This will also allow for ``while(){} else {}`` syntax.

```llvm
define dso_local i32 @main() #1 {
	%1 = icmp ult i32 0, i32 10
	br i1 %1, label %1

2:
	%3 = phi i32 [ %4, %3 ], [ %0, %1 ]
	call void @println.kyfn.g(i32 %2)
	%4 = add i32 %2, 1
	%5 = icmp ult i32 %5, 10
	br i1 %5, label %2, label %6

6:
	%7 = phi i32 [ %2, %1 ], [ %5, %4 ]
	call void @println.kyfn.g(i32 %7)
	ret i32 0
}
```

This clearly shows there is room for more improvement as the entry block could be replaced with an unconditional jump as the condition is resolvable at compile time.

**Comment by AjaniBilby** [comment]
Implemented in #47
